boot/env.ml

```diff
@@ -46,23 +50,40 @@ let guess_coqlib () =
 
 (* Build layout uses coqlib = coqcorelib *)
 let guess_coqcorelib lib =
-  if Sys.file_exists (Path.relative lib "plugins")
+  if Sys.file_exists (Path.relative lib plugins_dir)
   then lib
   else Path.relative lib "../coq-core"
 
+let fail_lib lib =
+  let open Printf in
+  eprintf "File not found: %s\n" lib;
+  eprintf "The path for Coq libraries is wrong.\n";
+  eprintf "Coq libraries are shipped in the coq-stdlib package.\n";
+  eprintf "Please check the COQLIB env variable or the -coqlib option.\n";
+  exit 1
+
+let fail_core plugin =
+  let open Printf in
+  eprintf "File not found: %s\n" plugin;
+  eprintf "The path for Coq plugins is wrong.\n";
+  eprintf "Coq plugins are shipped in the coq-core package.\n";
+  eprintf "Please check the COQCORELIB env variable.\n";
+  exit 1
+
+let validate_env ({ core; lib } as env) =
+  let lib = Filename.concat lib prelude in
+  if not (Sys.file_exists lib) then fail_lib lib;
+  let plugin = Filename.concat core plugins_dir in
+  if not (Sys.file_exists plugin) then fail_core plugin;
+  env
+
 (* Should we fail on double initialization? That seems a way to avoid
    mis-use for example when we pass command line arguments *)
 let init () =
   let lib = guess_coqlib () in
   let core = Util.getenv_else "COQCORELIB"
       (fun () -> guess_coqcorelib lib) in
-  { core ; lib }
-
-let init () =
-  let { core; lib } = init () in
-  (* debug *)
-  if false then Format.eprintf "core = %s@\n lib = %s@\n%!" core lib;
-  { core; lib }
+  validate_env { core ; lib }
```

Comment:
This is a bit strange, if `guess_coqlib` fails it will already do `exit 1`, so this is not going to be run, right @gares?